src/components/invite/useInvites.js

```diff
@@ -167,26 +174,55 @@ const useInvited = () => {
 
       log.debug('updateInvited got invitees profiles', { invited })
 
-      //keep if both name + avatar or not in cache
-      invited = invited.filter(_ => _.name || _.avatar || cached.find(c => c.addr === _.addr) === undefined)
+      // keep if both name + avatar or not in cache
+      invited.filter(({ name, avatar, addr }) => {
+        if (name || avatar) {
+          return true
+        }
 
-      //adding/updating invitees profiles to cache
+        return !find(cached, { addr })
+      })
+
+      // adding/updating invitees profiles to cache
       cached = uniqBy(invited.concat(cached), 'addr')
-      cached.forEach(i => (i.status = pending[i.addr] ? 'pending' : 'approved'))
-      setInvites(cached)
-      AsyncStorage.setItem('GD_cachedInvites', cached)
+
+      // calculcate statuses
+      cached = cached.map(item => {
+        const { addr } = item
+        const status = pending[addr] ? 'pending' : 'approved'
+
+        return {
+          ...item,
+          status,
+        }
+      })
+
+      setInvitees(cached)
+      userStorage.userProperties.set('cachedInvites', cached)
     } catch (e) {
       log.error('updateInvited failed:', e.message, e)
     }
-  }
+  }, [setInvitees])
 
   useEffect(() => {
-    updateInvited()
-  }, [])
+    const updateData = async () => {
+      try {
+        const user = await goodWallet.invitesContract.methods.users(goodWallet.account).call()
+        const level = await goodWallet.invitesContract.methods.levels(user.level).call()
+
+        setLevel(level)
+        setTotalEarned(user.totalEarned.toNumber() / 100) //convert from wei to decimals
+      } catch (e) {
+        log.error('updateData failed:', e.message, e)
+      }
+    }
 
-  const { pending = [], approved = [] } = groupBy(invites, 'status')
+    setLastState(defaults(userStorage.userProperties.get('lastInviteState') || {}, defaultLastInviteState))
```

Comment:
1. moved it here to manage both invitees cache and lastInviteState in a single place
2. removed extra AsynsStorage serialization, as userProperties already cached inside it